hyperspy/io_plugins/emd.py

```diff
@@ -404,14 +432,707 @@ def log_info(self):
+                lambda x: '{:.3f}'.format(float(x))),
+            'ImportedDataParameter.Frame_number': (
+                "Acquisition_instrument.TEM.Detector.EDS.frame_number", None)
+        }
+
+        # Add selected element
+        if map_selected_element:
+            mapping.update({'Operations.ImageQuantificationOperation': (
+                            'Sample.elements',
+                            self._convert_element_list),
+                            })
+
+        return mapping
+
+    def _convert_element_list(self, d):
+        atomic_number_list = d[d.keys()[0]]['elementSelection']
+        return [atomic_number2name[int(atomic_number)] for atomic_number in atomic_number_list]
+
+    def _convert_time(self, unix_time):
+        # Since we don't know the actual time zone of where the data have been
+        # acquired, we convert the datetime to the local time for convinience
+        dt = datetime.fromtimestamp(float(unix_time), tz=tz.tzutc())
+        return dt.astimezone(tz.tzlocal()).isoformat().split('+')[0]
+
+    def _get_local_time_zone(self):
+        return tz.tzlocal().tzname(datetime.today())
+
+
+class FeiSpectrumStreamContainer(object):
+
+    """
+    Container class to manage the different SpectrumStreams
+
+    When we read the array, we could add an option to import only X-rays
+    acquired within a "frame range" (specific scanning passes) or reading all
+    frames individually.
+    Each detector can also be read individually, otherwise the X-ray counts
+    are summed over all detectors.
+    """
+
+    def __init__(self, spectrum_stream_group, shape, energy_rebin=1,
+                 first_frame=0, last_frame=None, individual_frame=False,
+                 data_dtype=None, lazy=False):
+        self.spectrum_stream_group = spectrum_stream_group
+        self.shape = shape
+        self.energy_rebin = energy_rebin
+        self.first_frame = first_frame
+        self.last_frame = last_frame
+        self.individual_frame = individual_frame
+        self.data_dtype = data_dtype
+        self.lazy = lazy
+
+    def _read_all_streams(self, subgroup_keys):
+
+        stream_data_list = [
+            self.spectrum_stream_group['{}/Data'.format(key)][:].T[0]
+            for key in subgroup_keys]
+
+        streams = [self._read_individual_stream(stream_data, key)
+                   for key, stream_data in zip(subgroup_keys, stream_data_list)]
+
+        return streams
+
+    def _read_individual_stream(self, stream_data, key):
+        stream = FeiSpectrumStream(stream_data, **self.kwargs)
+
+        acquisition_key = '{}/AcquisitionSettings'.format(key)
+        acquisition_settings_group = self.spectrum_stream_group[acquisition_key]
+        stream.parse_acquisition_settings(acquisition_settings_group)
+
+        metadata_group = self.spectrum_stream_group[key]
+        stream.parse_metadata(metadata_group)
+#        stream._memory_check()
+
+        stream.get_spectrum_image()
```

Comment:
As the method doesn't return anything, what about `compute_spectrum_image` instead?